tests/integration/test_startup_scripts_execution_state/test.py

```diff
@@ -46,11 +51,40 @@ def test_startup_execution_state(start_cluster):
         == STATE_SUCCESS
     )
 
-    assert (
-        int(
-            bad.query(
-                "SELECT value FROM system.metrics WHERE metric = 'StartupScriptsExecutionState'"
-            ).strip()
-        )
-        == STATE_FAILURE
+    assert_startup_script_failed()
+    bad.stop_clickhouse()
+    # Set throw_on_error: true for the startup_script
+    bad.replace_in_config(
```

Comment:
See comment: https://github.com/ClickHouse/ClickHouse/pull/79732#discussion_r2078967153 - you can set it to false by default and then just replace `false` with `true` i.e. something like this:
```
 bad.replace_in_config("/etc/clickhouse-server/config.d/bad_script.xml",  "<throw_on_error>false",  "<throw_on_error>true")
```

Reply:
nice! fixed